### PR TITLE
move some of the inelastic scatter stuff up to C++

### PR DESCRIPTION
**Pull request by zingale**
this eliminates the need for state_indices in Fortran

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Source/radiation/MGFLD.cpp [hunk 1]
```diff
@@ -1670,13 +1670,41 @@ void Radiation::inelastic_scattering(int level)
...
+                    Real ks = kps_arr(i,j,k);
+
+                    Real dEr = 0.0;
+
+                    Real Er[NGROUPS];
+                    for (int g = 0; g < NGROUPS; ++g) {
+                        Er[g] = Er_arr(i,j,k,g);
+                    }
+
+                    ca_inelastic_sct(i, j, k,
```
Comment (maxpkatz) [inline Source/radiation/MGFLD.cpp:1694]: This call won't work in the GPU build; how much extra work is it to finish the GPU port of the inelastic scattering function?

Comment (zingale) [reply Source/radiation/MGFLD.cpp:1694]: it is ugly and sort of orthogonal to my goal here.  Maybe I'll just remove the GPU tag.

**Comment by zingale** [comment]
all tests pass:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-11-30-004/index.html

File: Source/radiation/MGFLD.cpp (hunk 1 above)
Comment (zingale) [reply Source/radiation/MGFLD.cpp:1694]: removed from the GPU for now

Comment (zingale) [reply Source/radiation/MGFLD.cpp:1694]: switched to `LoopOnCpu`

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**